numpydoc/cli.py

```diff
@@ -0,0 +1,68 @@
+"""The CLI for numpydoc."""
+
+import argparse
+import ast
+from typing import List, Sequence, Union
+
+from .docscrape_sphinx import get_doc_object
+from .hooks import validate_docstrings
+from .validate import Validator, validate
+
+
+def render_object(import_path: str, config: Union[List[str], None] = None) -> int:
+    """Test numpydoc docstring generation for a given object."""
+    # TODO: Move Validator._load_obj to a better place than validate
+    print(get_doc_object(Validator._load_obj(import_path), config=dict(config or [])))
+    return 0
+
+
+def validate_object(import_path: str) -> int:
+    """Run numpydoc docstring validation for a given object."""
+    exit_status = 0
+    results = validate(import_path)
+    for err_code, err_desc in results["errors"]:
+        exit_status += 1
+        print(":".join([import_path, err_code, err_desc]))
+    return exit_status
+
+
+def main(argv: Union[Sequence[str], None] = None) -> int:
+    """CLI for numpydoc."""
+    ap = argparse.ArgumentParser(prog="numpydoc", description=__doc__)
+    subparsers = ap.add_subparsers(title="subcommands")
+
+    def _parse_config(s):
+        key, _, value = s.partition("=")
+        value = ast.literal_eval(value)
+        return key, value
+
+    render = subparsers.add_parser(
+        "render",
+        description="Test numpydoc docstring generation for a given object.",
```

Comment:
Why does this "Test" docstring generation; doesn't it generate an expanded RST version of the docstring?

Reply:
This actually comes from the original docstring for `render_object()`, which was the only thing called for the original CLI's behavior:

https://github.com/numpy/numpydoc/blob/3a21e5440c90a0f306c42fdc303083a12927969b/numpydoc/__main__.py#L14

Reply:
:shrug: That's weird... but there are lots of weird things in this code :) Feel free to change whatever you see fit, or if it feels out of scope that's fine too.

Reply:
I will use a version of what you put above. This PR has lots of potential for snowballing, and I want to keep it as focused as possible.